pkg/resolve/resolve.go

```diff
@@ -45,8 +46,14 @@ func ImageReferences(input []byte, builder build.Interface, publisher publish.In
 		}
 		// This simply returns the replaced object, which we discard during the gathering phase.
 		if _, err := replaceRecursive(obj, func(ref string) (string, error) {
-			if builder.IsSupportedReference(ref) {
-				refs[ref] = struct{}{}
+			if strict && !strings.HasPrefix(ref, "ko://") {
```

Comment:
You could dry this up a bit -- reuse `tref` on line 56 and pull `strings.HasPrefix(ref, "ko://")` out into a boolean, maybe `strictRef`.

Reply:
Good call, done.